From a09450c1c5fb08acf1cf8944ed84af3fd0ac6785 Mon Sep 17 00:00:00 2001
From: Brian Behlendorf <behlendorf1@llnl.gov>
Date: Thu, 3 Dec 2015 10:27:24 -0800
Subject: [PATCH 01/18] Treat RETRY as FAILURE

When testing kernel modules with buildbot it's entirely possible that
the system may panic resulting in a disconnect.  For the purposes of
automated testing we want to consider this a FAILURE and immediately
move on to testing the next change.

This change should not be pushed upstream until the behavior is made
more flexible.  For example, this could be done using a 'flunkOnRetry'
build step option.

Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>
---
 master/buildbot/process/build.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/master/buildbot/process/build.py b/master/buildbot/process/build.py
index fc720d924..e0e3ffacd 100644
--- a/master/buildbot/process/build.py
+++ b/master/buildbot/process/build.py
@@ -454,7 +454,10 @@ class Build(properties.PropertiesMixin):
                 possible_overall_result = WARNINGS
             if step.flunkOnWarnings:
                 possible_overall_result = FAILURE
-        elif result in (EXCEPTION, RETRY):
+        elif result == RETRY:
+            possible_overall_result = FAILURE
+            terminate = True
+        elif result == EXCEPTION:
             terminate = True
 
         # if we skipped this step, then don't adjust the build status
-- 
2.14.3

